flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java

```diff
@@ -180,7 +183,7 @@ public WebMonitorEndpoint(
 		this.restConfiguration = Preconditions.checkNotNull(restConfiguration);
 		this.resourceManagerRetriever = Preconditions.checkNotNull(resourceManagerRetriever);
 		this.transientBlobService = Preconditions.checkNotNull(transientBlobService);
-		this.executor = Preconditions.checkNotNull(executor);
+		this.executor = Executors.newFixedThreadPool(endpointConfiguration.getNumThreads(), new ExecutorThreadFactory("Flink-" + getClass().getSimpleName()));
```

Comment:
This could be a follow up: Can we start the threads with `Thread.MIN_PRIORITY` in order to avoid congestion wrt the rpc threads?

Reply:
Shouldn't we be worried about the other end of worst-case scenarios: the REST API (and thus client and WebUI) becoming completely unresponsive?

Reply:
A simpler alternative might be to bump the Dispatcher priority instead; then we don't have to adjust every other component (like metrics).

Reply:
In general, I think the REST API is less important than the cluster components. Bumping the priority of the `Dispatcher` could also be a solution.